reconcile/utils/terraform_client.py

```diff
@@ -231,13 +231,16 @@ def get_terraform_output_secrets(self):
 
         return data
 
-    def populate_desired_state(self, ri, oc_map, tf_namespaces):
+    def populate_desired_state(self, ri, oc_map, tf_namespaces, account_name):
         self.init_outputs()  # get updated output
 
         # Dealing with credentials for RDS replicas
         replicas_info = self.get_replicas_info(namespaces=tf_namespaces)
 
         for account, output in self.outputs.items():
+            if account_name and account != account_name:
+                continue
+
```

Comment:
It'd be worth adding a test for this. According to the PR description this is pretty critical.